# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: startupcache/StartupCache.h
# Commit: a727a8fe08f5
# Full Hash: a727a8fe08f50746757656ba9599047dac6c0420
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-02-13 21:42:57
# Regressor Bug: 1607505
# File Overlap Count: 3
# Description:
#   Bug 1607505 - Start writing StartupCache earlier in shutdown r=froydnj a=reland CLOSED TREE
#   
#   The initial thought for getting the StartupCache out of the shutdown
#   path was to simply not write it during shutdown, as it should write
#   60 seconds after startup, and the theory was that if the user shut
# ==============================================================================

diff -r 752a464c8ec8 -r a727a8fe08f5 startupcache/StartupCache.h
--- a/startupcache/StartupCache.h	Thu Feb 13 16:56:44 2020 +0000
+++ b/startupcache/StartupCache.h	Wed Feb 12 19:02:12 2020 +0000
@@ -163,6 +163,10 @@
   // Removes the cache file.
   void InvalidateCache(bool memoryOnly = false);
 
+  // For use during shutdown - this will write the startupcache's data
+  // to disk if the timer hasn't already gone off.
+  void MaybeInitShutdownWrite();
+
   // Signal that data should not be loaded from the cache file
   static void IgnoreDiskCache();
 
@@ -203,6 +207,7 @@
   void WaitOnWriteThread();
   void WaitOnPrefetchThread();
   void StartPrefetchMemoryThread();
+  void MaybeSpawnWriteThread();
 
   static nsresult InitSingleton();
   static void WriteTimeout(nsITimer* aTimer, void* aClosure);